### ISSUE TITLE
importin `ipdb` is much slower than importing `pdb`

### ISSUE THREAD

**Issue by avatar-lavventura** [gotcha/ipdb#202]
I observe that importing `ipdb` takes much longer time than `pdb` almost 120 times on my end. During testing this will increase time to lose for each starting session. 

What may be the reason for this (is it a normal case)? Is there any way to make it faster to import `ipdb`

- importing `ipdb` (`ipdb==0.13.3`)

`0.60s user 0.09s system 90% cpu 0.754 total`

- importing `ipdb`

`0.05s user 0.01s system 97% cpu 0.062 total`

**Comment by gotcha** [comment]
Is this true for earlier versions ? There might be some recent change that is too slow...

**Comment by avatar-lavventura** [comment]
@gotcha I just started using `ipdb` with its latest version `0.13.3` (I really like it and it is a great tool). On `1080ti` I don't see much slow load time but when I try it on macOS Catalina Version 10.15.5 , the load time was much longer. I was just curios was it related to me or something else.

Adding `export PYTHONBREAKPOINT=ipdb.set_trace` into my `.bashrc` file and calling breakpoint() actually helped . It imports the library if and only if I use `breakpoint()` in a line rather than importing all the time.

**Comment by alexandrebarbaruiva** [comment]
@avatar-lavventura how much slower? I might test it on different versions out of curiosity

**Comment by alexandrebarbaruiva** [comment]
@avatar-lavventura how did you test the importing times?

**Comment by avatar-lavventura** [comment]
@alexandrebarbaruiva 

I am just using `time` command.

```
❯ time py _ipdb.py
python3 -B -tt _ipdb.py  0.71s user 0.08s system 98% cpu 0.806 total
❯
❯ time py _pdb.py
python3 -B -tt _pdb.py  0.06s user 0.01s system 96% cpu 0.079 total
```

---
The catch time till breakpoint is seems more due to importing for `ipdb` on me end.

`_pdb.py`:

```
import ipdb
ipdb.set_trace()
```

`_ipdb.py`:

```
import ipdb
ipdb.set_trace()

```

**Comment by adamchainz** [comment]
This is fixed by #224 which instantiates IPython lazily.

I timed with [hyperfine](https://github.com/sharkdp/hyperfine) which does repeated runs and provides statistics.

On master:

```
$ hyperfine "python -c 'import ipdb'"
Benchmark #1: python -c 'import ipdb'
  Time (mean ± σ):     469.1 ms ±  63.6 ms    [User: 386.2 ms, System: 64.4 ms]
  Range (min … max):   436.5 ms … 647.6 ms    10 runs
```

With #224 applied:

```
$ hyperfine "python -c 'import ipdb'"
Benchmark #1: python -c 'import ipdb'
  Time (mean ± σ):     369.6 ms ±   7.5 ms    [User: 310.7 ms, System: 55.8 ms]
  Range (min … max):   360.3 ms … 383.6 ms    10 runs
```

For reference here's the import time of just `IPython`:

```
$ hyperfine "python -c 'import IPython'"
Benchmark #1: python -c 'import IPython'
  Time (mean ± σ):     367.2 ms ±   5.9 ms    [User: 306.4 ms, System: 56.8 ms]
  Range (min … max):   360.5 ms … 381.2 ms    10 runs
```

**Comment by avatar-lavventura** [comment]
@adamchainz Thank you. Is the fix pushed into the master or should I fork the branch where the commit it?

**Comment by adamchainz** [comment]
I made a PR, the PR has not been merged.

**Comment by gotcha** [comment]
https://pypi.org/project/ipdb/0.13.7/ released with #224

@avatar-lavventura Would you pleased close the issue if fixed for you ?

**Comment by avatar-lavventura** [comment]
@gotcha Done. // @adamchainz Thanks for the fixed. I never heard `hyperfine` before, its a great tool
